builtin/logical/pki/ocsp_test.go

```diff
@@ -0,0 +1,542 @@
+package pki
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ocsp"
+)
+
+// If the ocsp_disabled flag is set to true in the crl configuration make sure we always
+// return an Unauthorized error back as we assume an end-user disabling the feature does
+// not want us to act as the OCSP authority and the RFC specifies this is the appropriate response.
+func TestOcsp_Disabled(t *testing.T) {
+	t.Parallel()
+	type testArgs struct {
+		reqType string
+	}
+	var tests []testArgs
+	for _, reqType := range []string{"get", "post"} {
+		tests = append(tests, testArgs{
+			reqType: reqType,
+		})
+	}
+	for _, tt := range tests {
+		localTT := tt
+		t.Run(localTT.reqType, func(t *testing.T) {
+			b, s, testEnv := setupOcspEnv(t, "rsa")
+			resp, err := CBWrite(b, s, "config/crl", map[string]interface{}{
+				"ocsp_disable": "true",
+			})
+			requireSuccessNilResponse(t, resp, err)
+			resp, err = sendOcspRequest(t, b, s, localTT.reqType, testEnv.leafCertIssuer1, testEnv.issuer1, crypto.SHA1)
+			require.NoError(t, err)
+			requireFieldsSetInResp(t, resp, "http_content_type", "http_status_code", "http_raw_body")
+			require.Equal(t, 401, resp.Data["http_status_code"])
+			require.Equal(t, "application/ocsp-response", resp.Data["http_content_type"])
+			respDer := resp.Data["http_raw_body"].([]byte)
+
+			require.Equal(t, ocsp.UnauthorizedErrorResponse, respDer)
+		})
+	}
+}
+
+// If we can't find the issuer within the request and have no default issuer to sign an Unknown response
+// with return an UnauthorizedErrorResponse/according to/the RFC, similar to if we are disabled (lack of authority)
```

Comment:
Might be worth noting this is the behavior for when an issuer is removed from a mount, unlike CRLs.

Reply:
Noting in documentation you mean?

Reply:
Nah, just in the test comment so we know that's intentional.